File: src/main/java/com/github/resources/ApResource.java
package com.github.resources;

import com.github.repository.ApRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

/**
 * Created by zk_chs on 16/4/11.
 */
@RestController
@RequestMapping(value = "/ap")
public class ApResource {

    @Autowired
    private ApRepository apRepository;

    @RequestMapping(value = "/ap", method = RequestMethod.GET)
    public Object get (@PathVariable(value = "id") int id){
        return apRepository.findOne(id);
    }

}

File: src/main/scala/com/github/resources/ApResource.scala
package com.github.resources

import com.github.repository.ApRepository
import com.github.utils.M
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation._

/**
  * Created by zk_chs on 16/4/13.
  */
@RestController
@RequestMapping(Array("/ap"))
class ApResource @Autowired() (private val apRepository: ApRepository) {

  @RequestMapping(method = Array(RequestMethod GET, RequestMethod POST))
  def get(@RequestParam(value = "id", required = false) id: Int) = {
    if ( id == 0 ) M builder() put ("status", "fail") put ("msg", "id may not be null") buildMap()
    else M builder() put ("status", "success") put ("result", apRepository findOne id) buildMap()
  }

}
